akka-bench-jmh/src/main/scala/akka/stream/ZipWithIndexBenchmark.scala

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (C) 2018-2023 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.stream
+
+import akka.actor.ActorSystem
+import akka.stream.scaladsl._
+import com.typesafe.config.ConfigFactory
+import org.openjdk.jmh.annotations._
+
+import java.util.concurrent.TimeUnit
+import scala.collection.immutable
+import scala.concurrent.Await
+import scala.concurrent.duration._
+
+object ZipWithIndexBenchmark {
+  final val OperationsPerInvocation = 100000
+}
+
+@State(Scope.Benchmark)
+@OutputTimeUnit(TimeUnit.SECONDS)
+@BenchmarkMode(Array(Mode.Throughput))
+class ZipWithIndexBenchmark {
+  import ZipWithIndexBenchmark._
+
+  private val config = ConfigFactory.parseString("""
+    akka.actor.default-dispatcher {
+      executor = "fork-join-executor"
+      fork-join-executor {
+        parallelism-factor = 1
+      }
+    }
+    """)
+
+  private implicit val system: ActorSystem = ActorSystem("ZipWithIndexBenchmark", config)
+
+  @TearDown
+  def shutdown(): Unit = {
+    Await.result(system.terminate(), 5.seconds)
+  }
+
+  private val newZipWithIndex =
+    Source.repeat(1).take(OperationsPerInvocation).zipWithIndex.toMat(Sink.ignore)(Keep.right)
```

Comment:
You could use `BenchTestSourceSameElement` for less overhead of irrelevant things. You can see that in several other benchmarks. As Sink I think we use LatchSink for the same reason.